micronaut-mcp-server-java-sdk/src/main/java/io/micronaut/mcp/server/json/JsonSchemaValidatorReplacement.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.mcp.server.json;
+
+import com.networknt.schema.SpecVersion;
+import com.networknt.schema.AbsoluteIri;
+import com.networknt.schema.InputFormat;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.SchemaValidatorsConfig;
+import com.networknt.schema.ExecutionContextCustomizer;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.resource.InputStreamSource;
+import com.networknt.schema.resource.SchemaLoader;
+import io.micronaut.context.annotation.Replaces;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.io.ResourceLoader;
+import io.micronaut.json.JsonMapper;
+import io.micronaut.jsonschema.utils.JsonSchemaClassPathResourceLoader;
+import io.micronaut.jsonschema.utils.JsonSchemaConfiguration;
+import io.micronaut.jsonschema.validation.JsonSchemaValidator;
+import io.micronaut.jsonschema.validation.JsonSchemaValidatorConfiguration;
+import io.micronaut.jsonschema.validation.ValidationMessage;
+import jakarta.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import io.micronaut.jsonschema.validation.ValidationMessageAdapter;
+
+@Replaces(JsonSchemaValidator.class)
+@Singleton
+@Internal
+final class JsonSchemaValidatorReplacement implements JsonSchemaValidator {
```

Comment:
why do we need a replacement? Is there some default that configures Jackson?

Reply:
resolved via https://github.com/micronaut-projects/micronaut-mcp/pull/52/commits/2558df74f7f236cb379fb8d02537984cff3e5ad5